erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -511,6 +511,17 @@ frappe.ui.form.on('Payment Entry', {
 		});
 	},
 
+	allocate_payment_amount: function(frm) {
+		if(frm.doc.references.length == 0){
+			frm.events.get_outstanding_documents(frm);
+		}
+
+		if(frm.doc.payment_type == "Pay")
+			frm.events.allocate_party_amount_against_ref_docs(frm, frm.doc.received_amount);
+		else
```

Comment:
if..else is not required, as we are returning if payment type is Internal Transfer